compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -936,6 +936,17 @@ pub fn prohibit_assoc_ty_binding(
             None
         },
     });
+
+    if let Some((segment, _)) = segment
+        && segment.args().parenthesized == hir::GenericArgsParentheses::No
+        && let hir::TypeBindingKind::Equality { term: hir::Term::Ty(ty) } = binding.kind
```

Comment:
It should quite easily be possible to generalize this to cover associated const equality constraints as well.

Reply:
I am covering consts as well now.